api/models/graphite_render_meta.go

```diff
@@ -0,0 +1,62 @@
+package models
+
+import (
+	"strconv"
+	"time"
+)
+
+// ResponseWithMeta is a graphite render response with metadata
+type ResponseWithMeta struct {
+	Meta   RenderMeta
+	Series SeriesByTarget
+}
+
+func (rwm ResponseWithMeta) MarshalJSONFast(b []byte) ([]byte, error) {
+	b = append(b, `{"meta-version":"v0.1","meta":`...)
```

Comment:
Why put this outside of the "meta" object in the response?

Reply:
it allows us to change how we put metadata into the response. maybe in the future the key won't be "meta" anymore, but something else.

Reply:
you've just created the same problem with `meta-version` though, one option might be to use an `X-` http header instead.

Reply:
Right, if you've got to pin a key, might as well just pick one. @DanCech - Do you mean using a header specifically for the meta version? I think it's simple enough to have the `version` in the `meta` object. *Everything* under the meta object can change between version, but `response["meta"]["version"]` should be a version string. The only issue here is if we ever need `"meta"` to mean something else that couldn't be a sub-object of `meta` itself, which seems unlikely to me.

Reply:
@shanson7 i don't understand how `response["meta"]["version"]` (2 levels into the response) is more robust than 1 level into the body, especially since we may want to rename the meta section itself.
I like dan's suggestion to use a header, however:
1) x- prefix is deprecated (https://tools.ietf.org/html/rfc6648)
2) i'm concerned that proxies may strip the header, it seems to me including the version in the http response body is a more robust way of making sure the information makes it back to the client
3) my intention was to version just the render response. putting the version in a header seems to imply something broader like versioning the api itself, which we want to do, but requires more thought.

Reply:
I don't see how number of levels effects robustness at all. Either way it's a path that can't (easily) be changed. It's just cleaner looking IMO to keep them together in a single object.

Hypothetically, if you rename "meta" it's going to be awfully odd to have "meta-version".

Reply:
to clarify, the version identifier applies to the response body as a whole, not [only] the particular meta section being introduced here. i envision in the future we will add stuff to the series dictionaries as well, for example.

Reply:
Ah, I see. Then I definitely would call it something else. Like just `"version"` or `"format-version"`. I definitely thought it was strictly associated with `"meta"` due to the name.